src/gauge.js

```diff
@@ -261,75 +275,196 @@
-            pathTransition = "stroke " + dur + "ms ease";
-            // textTransition = "fill " + dur + "ms ease";
+      /**
+       * Create single gauge section element
+       * @param {Object} v
+       * @param {Number} totalAngle
+       * @return {Number}
+       */
+      function updateSingleGaugeSection(v, totalAngle) {
+        var valueInPercentage = getValueInPercentage(v.value, min, limit),
+          // angle = getAngle(val, 360 - Math.abs(endAngle - startAngle)),
+          angle = getAngle(valueInPercentage, 360 - Math.abs(startAngle - endAngle)),
+          // this is because we are using arc greater than 180deg
+          flag = angle <= 180 ? 0 : 1;
+
+        var gaugeValuePath = svg("path", {
+          "class": valueDialClass,
+          fill: "none",
+          stroke: "#666",
+          "stroke-width": 2.5,
+          d: pathString(radius, startAngle, startAngle) // value of 0
+        });
+
+        gaugeElement.appendChild(gaugeValuePath);
+
+        if(imposeSection) {
+          gaugeValuePath.setAttribute(
+            "d",
+            pathString(radius, startAngle, angle + startAngle, flag)
+          );
+        } else {
+          gaugeValuePath.setAttribute(
+            "d",
+            pathString(radius, startAngle + totalAngle, angle + startAngle + totalAngle, flag)
+          );
+        }
+
+        updateColor(gaugeValuePath, v.color, 2);
+        return totalAngle + angle;
+      }
+
+      /**
+       * Update gauge data
+       * @param {Array} value
+       * @param {Number} frame
+       * @param {Number} sectionStartPoint - this value need for calculate start section point
+       */
+      function updateGauge(value, frame, sectionStartPoint) {
+
+        var label;
+
+        // Remove old gauge sections
+        clearGaugeSections();
+
+        // Check sectionStartPoint
+        sectionStartPoint = sectionStartPoint || 0;
+
+        // Create single gauge section
+        value.forEach(function (v) {
+          label = generateMultipleLabel(label, v.value);
+          sectionStartPoint = updateSingleGaugeSection(v, sectionStartPoint);
+        });
+        renderLabel(label);
+      }
+
+      function updateColor(el, color, duration) {
+        var dur = duration * 1000,
+          pathTransition = "stroke " + dur + "ms ease";
+
+        el.style = [
+          "stroke: " + color,
+          "-webkit-transition: " + pathTransition,
+          "-moz-transition: " + pathTransition,
+          "transition: " + pathTransition
+        ].join(";");
+      }
+
+      /**
+       * Remove all gauge sections
+       * @return {boolean}
+       */
+      function clearGaugeSections() {
+        var elements = gaugeElement.getElementsByClassName(defaultOptions.valueDialClass);
+        if (!elements.length) return false;
+        do {
+          gaugeElement.removeChild(elements[0]);
+        } while (elements.length);
+      }
+
+      function setGaugeColor(value, duration) {
```

Comment:
`setGaugeColor` and `updateColor` looks like the same methods, please leave only one

Reply:
Done